Hanson/Event Publisher/EventPublisher.swift

```diff
@@ -48,24 +50,32 @@ public protocol EventPublisher: class {
 }
 
 public extension EventPublisher {
-    
+
     func publish(_ event: Event) {
         lock.lock()
         defer { lock.unlock() }
-        
-        eventHandlers.forEach { _, eventHandler in
-            eventHandler(event)
+
+        eventHandlers.forEach { _, value in
+            let (eventHandler, eventScheduler) = value
+            eventScheduler.scheduleEvent {
+                eventHandler(event)
+            }
         }
     }
-    
+
     @discardableResult
     func addEventHandler(_ eventHandler: @escaping EventHandler<Event>) -> EventHandlerToken {
+        addEventHandler(eventHandler, eventScheduler: CurrentThreadScheduler())
+    }
+
+    @discardableResult
+    func addEventHandler(_ eventHandler: @escaping EventHandler<Event>, eventScheduler: EventScheduler) -> EventHandlerToken {
```

Comment:
I think we can have just one function with a default value, then we can remove the function on line 67.
```suggestion
    func addEventHandler(_ eventHandler: @escaping EventHandler<Event>, eventScheduler: EventScheduler = CurrentThreadScheduler()) -> EventHandlerToken {
```

Reply:
Ah yes of course, that's right. I'll update this, thanks!